jack.py

```diff
@@ -600,6 +680,14 @@ def transport_locate(self, frame):
         _check(_lib.jack_transport_locate(self._ptr, frame),
                "Error locating JACK transport")
 
+    def transport_query(self):
+        """return current transport position."""
+        # Test rolling transport status
+        if _lib.jack_transport_query(self._ptr, self.position) == 1:
+            return self.position
+        else:
+            return None
```

Comment:
Le 13/04/2015 20:24, Matthias Geier a écrit :
>
> In jack.py 
> <https://github.com/spatialaudio/jackclient-python/pull/10#discussion_r28264312>:
>
> > @@ -600,6 +680,14 @@ def transport_locate(self, frame):
> >          _check(_lib.jack_transport_locate(self._ptr, frame),
> >                 "Error locating JACK transport")
> >
> > +    def transport_query(self):
> > +        """return current transport position."""
> > +        # Test rolling transport status
> > +        if _lib.jack_transport_query(self._ptr, self.position) == 1:
> > +            return self.position
> > +        else:
> > +            return None
>
> I just saw that the return value isn't an error code but probably 
> useful information in a |jack_transport_state_t|.
> What about returning that, too?
>
Yes I think better is to return a tuple ? :

	return_value = _lib.jack_transport_query(..., self.position)
         return (return_value, self.position)

  Maybe there is interesting values in position when transport is stopped
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/spatialaudio/jackclient-python/pull/10/files#r28264312>.
>